### PR TITLE
Error handling and AuthServer stateless refactor

### LINKED ISSUE #71: Test authentication server in practise
We need to test authentication server in practise, full use case with webapp.

### LINKED ISSUE #101: DOESNT WORK

### PR DESCRIPTION
**Pull request by stanbar**
make AuthServerModule stateless (make serverCredentials local variable)
hasInput == is not null and is not empty
autolock listener
fixed config endpoint spelling bug
moved error handling to server
throws RuntimeExceptions in lambdas
replace runtime IllegalArgumentException with checked IncorrectPasswordException
update webapp accordingly
closes #101 
closes #71

Took 2 hours 44 minutes

### CODE REVIEW

File: Manager/src/main/java/io/raspberrywallet/manager/Manager.java [hunk 1]
```diff
@@ -69,17 +71,21 @@
...
-
-        Runnable onLockTriggered = () -> {
+        onLockTriggered = () -> {
             frontendChannel.info("Autolock triggered");
             try {
                 lockWallet();
             } catch (WalletNotInitialized ignored) {
                 //we don't care about locking if it wasn't even inited
+            } catch (IncorrectPasswordException | IOException e) {
+                throw new RuntimeException(e);
```
Comment (PatrykMilewski) [inline Manager/src/main/java/io/raspberrywallet/manager/Manager.java]: should it throw runtime exception? If this contructor is called on unlocking Wallet after power on, then it shouldn't be runtime exception

File: Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java [hunk 2]
```diff
@@ -193,7 +199,8 @@ private void synchronizeWalletNonBlocking(final Wallet wallet, @Nullable String
                 try {
                     saveEncryptedWallet(password);
                     Logger.info("Wallet balance" + wallet.getBalance().toFriendlyString());
-                } catch (IOException | WalletNotInitialized e) {
+                } catch (IOException | WalletNotInitialized | IncorrectPasswordException e) {
```
Comment (PatrykMilewski) [inline Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java]: IncorrectPasswordException should be separated from others and doesn't print stack trace

File: Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java [hunk 3]
```diff
@@ -208,10 +215,10 @@ public void onSuccess(@Nullable Object result) {
                     @Override
                     protected void progress(double pct, int blocksSoFar, Date date) {
                         super.progress(pct, blocksSoFar, date);
-                        Logger.d("Progress " + pct + "%");
+                        Logger.d("Progress " + (int) Math.round(pct) + "%");
```
Comment (PatrykMilewski) [inline Manager/src/main/java/io/raspberrywallet/manager/bitcoin/Bitcoin.java]: does it work correctly? (doule to integer casting problem on frontend, that was causing progress bar jumping
we were talking about that, not sure if you remember

File: ServerHttp/src/main/java/io/raspberrywallet/contract/IncorrectPasswordException.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package io.raspberrywallet.contract;
+
+public class IncorrectPasswordException extends Exception {
```
Comment (PatrykMilewski) [inline ServerHttp/src/main/java/io/raspberrywallet/contract/IncorrectPasswordException.java:3]: other exceptions in contract are extending Throwable class (not sure why), shouldn't we follow this?

**Review by PatrykMilewski — CHANGES_REQUESTED** [review]

**Comment by PatrykMilewski** [comment]
Also, I would't say, that AuthServerModule is stateless, since it hold session token and can be not / registered

**Comment by stanbar** [comment]
Right, the stateless was generalization here, I meant the password to be parsed each time from input and not cached in global field

**Review by PatrykMilewski — APPROVED** [review]

**Review by PatrykMilewski — APPROVED** [review]

**Comment by PatrykMilewski** [comment]
@stasbar Merge when done

and check Messenger (fb)

**Pull request merged**